core/matrix/dense.cpp

```diff
@@ -462,31 +462,30 @@ void Dense<ValueType>::compute_norm1_impl(LinOp* result) const
 template <typename ValueType>
 void Dense<ValueType>::convert_to(Dense<ValueType>* result) const
 {
-    if (this->get_size() && result->get_size() == this->get_size()) {
-        // we need to create a executor-local clone of the target data, that
-        // will be copied back later.
-        auto exec = this->get_executor();
-        auto result_array = make_temporary_output_clone(exec, &result->values_);
-        // create a (value, not pointer to avoid allocation overhead) view
-        // matrix on the array to avoid special-casing cross-executor copies
-        auto tmp_result =
-            Dense{exec, result->get_size(),
-                  Array<ValueType>::view(exec, result_array->get_num_elems(),
-                                         result_array->get_data()),
-                  result->get_stride()};
-        exec->run(dense::make_copy(this, &tmp_result));
-    } else {
-        result->values_ = this->values_;
-        result->stride_ = this->stride_;
-        result->set_size(this->get_size());
+    if (result->get_size() != this->get_size()) {
```

Comment:
If the original size is larger than required, perhaps we should preserve the original allocation. Then the new stride would be the original stride, while the new size is the new smaller size.

Reply:
Not sure about this. It might encourage users to keep pointers to the original allocation, which might cause issues if we change this behavior in the future.